Allow passing multiple columns to subset (#2926)
JuliaData/DataFrames.jl: docs/src/man/working_with_dataframes.md

   2 │      4
```

> The [`subset`](@ref) was designed in a way that is
> consistent how column transformations are specified in functions like

Reviewer: ```suggestion
The [`subset`](@ref) function has been designed in a way that is
consistent with how column transformations are specified in functions like
```

[`combine`](@ref), [`select`](@ref), and [`transform`](@ref). Examples of column
transformations accepted by these functions are provided in the following
section.

Additionally DataFrames.jl extends the [`filter`](@ref) and [`filter!`](@ref)

> functions provided in Julia Base and they also allow to subset a data frame.

Reviewer: ```suggestion
functions provided in Julia Base, which also allow subsetting a data frame.
```

> The use of these two functions is discouraged as they have an API that is not
> consistent with other transformation functions in DataFrames.jl. They are kept
> in the API for legacy reason, but it is recommended to use the [`subset`](@ref)
> and [`subset!`](@ref) functions instead.

Reviewer: I find this a bit strong: if that was the case we would deprecate them. Maybe something like this?
```suggestion
These methods are defined so that data frames implement the Julia API
for collections, but it is generally recommended to use the [`subset`](@ref)
and [`subset!`](@ref) functions instead, as they are both faster and
more consistent with other DataFrames functions.
```
